content/docs/arvo/concepts/scry.md

```diff
@@ -0,0 +1,77 @@
+
+To explore what scry endpoints are available for vanes, you can refer to the Scry Reference section of each vane in the [Arvo](/docs/arvo/overview) section of the documents.
+
+#### Gall agents
+
+Gall has a single scry endpoint of its own to check for the existence of an agent, but otherwise all Gall scries are passed through to one of the agents it manages. The target agent to scry is specified in place of the `desk` as described in the diagram above. Each Gall agent includes a `+on-peek` arm that defines its own scry endpoints. For example, `%graph-store` has a number of scry endpoints to access the data it stores, such as chat messages and the like.
+
+Gall agents can expose scry endpoints with any `care`, but most commonly they'll take a `%x` `care`. Gall handles `%x` scries specially - it expects an extra field at the end of the `path` that specifies a `mark`. Gall will attempt to perform a `mark` conversion from the `mark` returned by the scry endpoint to the `mark` specified.
+
+### What is an endpoint?
+
+"Endpoint" refers to a specific scry path in a vane or agent. They will sometimes informally be noted in documentation or source comments like `/x/foo/bar/baz` or maybe just `/foo/bar/baz`. The first part of the former example is the `care`, then the rest is the `path` portion as noted in the diagram earlier.
+
+If an agent's scry endpoints don't have formal documentation, you may need to refer to the `+on-peek` arm in its source code to determine its endpoints. While there's no exact pattern that's consistent across different agents, the general pattern will be some initial tests and conversions followed by a `?+` rune, whose cases will typically correspond to the endpoints. The `+on-peek` arm always return a `(unit (unit cage))` (a `cage` is a cell of `[mark vase]`), so each endpoint will typically finish by composing such a structure.
+
+Here's an example from the `+on-peek` arm of `%graph-store`:
+
+```hoon
+  [%x %keys ~]
+:-  ~  :-  ~  :-  %graph-update-2
+!>(`update:store`[now.bowl [%keys ~(key by graphs)]])
+```
+
+The case in the beginning says it takes a `%x` `care` and has a `path` of `/keys`. Additionally, it returns data with a `mark` of `%graph-update-2`. Assuming we wanted the `%graph-update-2` `mark` converted to a `%noun` `mark`, our scry would be composed along the lines of:
+
+```hoon
+.^(update:graph-store %gx /(scot %p our)/graph-store/(scot %da now)/keys/noun)
```

Comment:
This example is kind of silly in that the type given here is the type of the original mark. Might make the mark conversion more obvious by doing `.^(* ...` instead. (But even then, expecting a noun is always valid regardless of the mark.)